matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/sync/handler/SyncResponsePostTreatmentAggregatorHandler.kt

```diff
@@ -68,4 +76,21 @@ internal class SyncResponsePostTreatmentAggregatorHandler @Inject constructor(
             updateUserAccountDataTask.execute(UpdateUserAccountDataTask.DirectChatParams(directMessages = directChats))
         }
     }
+
+    private suspend fun fetchAndUpdateUsers(usersToFetch: List<String>) {
+        val userProfiles = fetchUsers(usersToFetch)
+        userProfiles.forEach { saveUserLocally(monarchy, it) }
+    }
+
+    private suspend fun fetchUsers(usersToFetch: List<String>) = usersToFetch.map {
+        val profileJson = profileService.getProfile(it)
+        User.fromJson(it, profileJson)
+    }
+
+    private fun saveUserLocally(monarchy: Monarchy, user: User) {
+        val userEntity = UserEntityFactory.create(user)
+        monarchy.doWithRealm {
+            it.insertOrUpdate(userEntity)
+        }
+    }
```

Comment:
It's good to split code into several smaller methods, but it's maybe harder to read in this particular case.

What about replacing those 3 methods with a single one:

```kotlin
    private suspend fun fetchAndUpdateUsers(usersToFetch: List<String>) {
        usersToFetch
                .map { userId ->
                    val profileJson = profileService.getProfile(userId)
                    User.fromJson(userId, profileJson)
                }
                .map { user -> UserEntityFactory.create(user) }
                .let { userEntities ->
                    monarchy.doWithRealm {
                        it.insertOrUpdate(userEntities)
                    }
                }
    }
```

Also has the advantage to do on single Realm transaction instead of N transactions with your code.

WDYT?